zebedee-cms/src/main/java/com/github/onsdigital/zebedee/filters/AuthenticationFilter.java

```diff
@@ -66,18 +100,65 @@ public boolean filter(HttpServletRequest request, HttpServletResponse response)
             return true;
         }
 
-        // Pass through without authentication for login requests:
-        // Password requests check
         Path path = Path.newInstance(request);
+    
         if (noAuthorisationRequired(path)) {
             return true;
         }
 
         // Check all other requests:
+        boolean result = false;
+        if (jwtSessionsEnabled) {
+            result = verifyAndStoreAccessToken(request, response);
+        } else {
+            result = processSession(request, response);
+        }
+        return result;
+    }
+
+    /**
+     * verifyAndStoreAccessToken - decodes, verifies and stores Access Token JWT
+     *
+     * @param request
+     * @param response
+     * @return boolean 
+     */
+    private boolean verifyAndStoreAccessToken(HttpServletRequest request, HttpServletResponse response) {
         boolean result = false;
         try {
-            Session session = Root.zebedee.getSessions().get(request);
+            //ensure that authorisation header present
+            if (request.getHeader("Authorization") == null) {
+                unauthorisedRequest(response, NO_AUTH_HEADER_FOUND);
+            } else {
+                try {
+                    sessions.set(request.getHeader("Authorization"));   
+                    result = true;
+                } catch (SessionsStoreException e) {
+                    // treat access token expired or malformed access token as unauthorised
+                    if (e.getMessage() == ACCESS_TOKEN_EXPIRED_ERROR || e.getMessage() == TOKEN_NOT_VALID_ERROR ) {
+                        unauthorisedRequest(response, e.getMessage());
+                    } else {
+                        accessTokenError(response, e.getMessage());
+                    }
+                }
+            }
+        } catch (IOException e) {
```

Comment:
See comment below for `unauthorisedRequest`. If you implement the suggestion or something similar it will remove the need to have these try-catch blocks for this IOException.